src/pluggy/callers.py

```diff
@@ -205,4 +212,8 @@ def _multicall(hook_impls, caller_kwargs, firstresult=False):
             except StopIteration:
                 pass
 
+        if with_impl:
```

Comment:
im a bit torn on this one
i wonder if enabling having the ourcome instead of the result as output could help to make this more nice

i also wonder how/if we could track force_result based hookwrapper changes

Reply:
Thanks for taking a look @RonnyPfannschmidt

Can you elaborate a bit what you mean by “enabling having the outcome instead of the result as output”?

Yeah this does ignore wrapper based changes at the moment (Is that what you mean by outcome vs result?).  I can think harder about how we might track that, but it would obviously require hacking the `_Result` object more. We could make the returned hookimpl a list instead of a single value, and throw in all the wrappers that modified the result?  

(Personally I’m ok with omitting wrapper logic here and just pointing it out in the docs)

Reply:
@tlambert03 basically the idea would be that outcome also tracks the hooks that add results, and is returned on parameter choice,

i'm a bit torn here what's the best way to give the feature a good expression, it may be also more helpful to get the belonging plugin object instead of the hook object 

the problem with this feature proposal is that i don't have a good impression/understanding of the use-case you want to enable, that makes misunderstanding very easy due o the XY problem

Reply:
thanks again for your time/input.

> basically the idea would be that outcome also tracks the hooks that add results, and is returned on parameter choice.

I'm afraid I'm still confused by your wording...  This PR _does_ track the hookimpls that add the results by optionally returning results as 2-tuples: `[(result, hookimpl), (result, hookimpl), ...]`.  If we want to extend that to also detect when hookwrappers use `_Result.force_result`, I could do this:
- add a `_was_forced = False` attribute to the `_Result` object
- when `_Result.force_result` is called, set `self._was_forced = True`
- in the [`_multicall` post-yield teardown loop](https://github.com/pytest-dev/pluggy/blob/6e8b6d4e8e0519e50af05b4e4b90b18b924bcbcc/src/pluggy/callers.py#L201-L206), check whether `outcome._was_forced` is true after each `gen.send()` call
  - if so, reset `outcome._was_forced` to `False` and add the corresponding hookwrapper to the list of returned objects.

obviously, this doesn't tell us _which_ results the hookwrapper clobbered, but I don't think we should go down the path of doing pre-post comparisons to see what/whether the hookwrapper changed, since that could be buggy and cause overhead.

> i'm a bit torn here what's the best way to give the feature a good expression, it may be also more helpful to get the belonging plugin object instead of the hook object

I have a slight preference for the `hookimpl` object, since a) it's the hookimpl that actually returned the result b) the plugin object is easily accessible anyway at `hookimpl.plugin`, but c) getting to specific hookimpl object that returned the result from the plugin object would require another step.

> the problem with this feature proposal is that i don't have a good impression/understanding of the use-case you want to enable, that makes misunderstanding very easy due o the XY problem

I updated the original post above with a full use case.  Primary goal here is to exit the hook chain as fast as possible (with `firstresult`) while _also_ knowing how you exited the hook chain (i.e. knowing which hookimpl returned non-None)